frontend/src/components/drawer/Drawer.tsx

```diff
@@ -52,19 +65,48 @@ const useStyles = makeStyles((theme: Theme) =>
 interface DrawerProps {
   navigationDrawerOpen: boolean;  // whether or not Drawer comp is displayed
   handleDrawerClose(): void;
+  handleUserSignOut(): void;
 }
 
 export default function DrawerComp(props: DrawerProps) {
   const classes = useStyles();
   const theme = useTheme();
 
+  const contextServices = React.useContext(ServiceContext);
+  const userLoginStatusService = React.useContext(UserLoginStatusContext);
+
+  const [displaySignOutWindow, setDisplaySignOutWindow] = React.useState<boolean>(false);
+
   // Associates each URL with a text representation and icon
   const pagesList = [
     {text: "Your Clubs", url: "/YourClubs", icon: <LibraryBooksIcon />},
     {text: "Explore", url: "/Explore", icon: <ExploreIcon />},
     {text: "Profile", url: "/Profile", icon: <FaceIcon />}
   ];
 
+  const handleOpenSignOutWindow = () => {
+    setDisplaySignOutWindow(true);
+  }
+
+  const handleCloseSignOutWindow = () => {
+    setDisplaySignOutWindow(false);
+  }
+
+  const handleSignOut = async () => {
+    const authenticationHandlerService = contextServices.authenticationHandlerService;
+    const loginStatusHandlerService = userLoginStatusService.loginStatusHandlerService;
+
+    loginStatusHandlerService.setUserLoginStatus(false);
+    loginStatusHandlerService.setUserToken(undefined);
```

Comment:
nit: Setting the login status to false should probably automatically clear out the user token. In general, services like this should not expose functionality that makes it possible to put them in an invalid state.